sequila/sequila-core/src/physical_planner/joins/interval_join.rs

```diff
@@ -1046,20 +1044,28 @@ impl IntervalJoinStream {
         let start = evaluate_as_i32(self.right_interval.start(), &state.batch)?;
         let end = evaluate_as_i32(self.right_interval.end(), &state.batch)?;
 
-        let mut left_builder = UInt32BufferBuilder::new(0);
-        let mut right_builder = UInt32BufferBuilder::new(0);
+        let mut builder_left = PrimitiveBuilder::<UInt32Type>::new();
 
+        let mut rle_right: Vec<u32> = Vec::with_capacity(self.hashes_buffer.len());
```

Comment:
I have come up with this solution, which is better then what we have and it seems like it gives slightly better performance then yours:
```rust
        // pre-allocate 2 vectors of at least the batch size
        let mut left_indexes: Vec<u32> = Vec::with_capacity(self.hashes_buffer.len());
        let mut right_indexes: Vec<u32> = Vec::with_capacity(self.hashes_buffer.len());

        for (i, hash_val) in self.hashes_buffer.iter().enumerate() {
            build_side
                .hash_map
                .get(*hash_val, start.value(i), end.value(i), |pos| {
                    left_indexes.push(pos as u32);
                    right_indexes.push(i as u32);
                })
        }

        let left_indexes: UInt32Array = left_indexes.into(); 
        let right_indexes: UInt32Array = right_indexes.into();
```

Reply:
Let's benchmark it!